src/thor/bidirectional_astar.cc

```diff
@@ -95,8 +96,8 @@ void BidirectionalAStar::Init(const PointLL& origll, const PointLL& destll) {
   // searches.
   cost_diff_ = mincostf - mincostr;
 
-  // Initialize best connection with max cost
-  best_connection_ = {GraphId(), GraphId(), std::numeric_limits<float>::max()};
+  // Initialize best connections as having none
+  best_connections_ = {};
```

Comment:
this is the main change to the algorithm. instead of keeping track of one connection, we keep track of the a set of the best connections.